packages/mobile/src/services/offline-downloader/offline-downloader.ts

```diff
@@ -0,0 +1,102 @@
+import path from 'path'
+
+import type { Track, UserMetadata } from '@audius/common'
+import {
+  encodeHashId,
+  accountSelectors,
+  cacheTracksSelectors
+} from '@audius/common'
+import { uniq } from 'lodash'
+import RNFS, { exists } from 'react-native-fs'
+
+import { store } from 'app/store'
+
+import { apiClient } from '../audius-api-client'
+
+import {
+  getLocalAudioPath,
+  getLocalCoverArtPath,
+  getLocalTrackJsonPath
+} from './offline-storage'
+const { getUserId } = accountSelectors
+const { getTrack } = cacheTracksSelectors
+
+// Main entrypoint - perform all steps required to complete a download
+export const downloadTrack = async (trackId: number, collection: string) => {
+  const state = store.getState()
```

Comment:
I don't think we are doing this anywhere else, all this logic feels like it might actually belong in sagas. But I'm pretty anti-saga these days so would love to hear other people's thoughts

Reply:
It felt weird writing it too. Currently, we usually call this from a component that has the state, but I wanted to keep it reusable for the potential background process. However, we won't be able to use the state their either. Keeping it simple for now, but open to ideas.

Reply:
Discussed in standup. Seems like this is the way for now